doc/contributions/coding_style.md

```diff
@@ -0,0 +1,8 @@
+# Coding style
+
+TODO: what is the coding style?
```

Comment:
We could use clang-format to enforce this and add a CI job to check each PR.

Reply:
> Personally I'd prefer people read and internalize the guidance and follow it when writing and reviewing.

But some people won't do it, so what's the purpose of enforcing its usage if we aren't checking it?

> Then again, that's a lot to ask of people, and CI can be help. So as long as it's clear that that CI job is there to help and not to be the decisive judge, it'd be useful.

In the end, it's up to the maintainer to decide if a given rule can be broken in some circumstances. If it's fine, they can decide to merge a PR even if the coding style rules are failing. And that's fine, all these CI checks are there to help us.